### PR TITLE
Uranium eating aegun, second try

### PR DESCRIPTION
**Pull request by 00ze-cyclone**
# Document the changes in your pull request

Aegun being just egun+ is boring and it being printable by the billion make it the absolute no brainer pick once it is researched.
There an entire system to make it give you 80 rads after being EMPed 73720372 times which unsurprisingly never fucking happen so that whole code is pretty much useless.

Now aegun is recharged with uranium instead of passively charging, the overloading bullshit that didn't matter ever got yeeted.
You use a sheet of uranium on the aegun and wait one second to refill it by 50%, doing so give you 150 rads so you have to worry about that (also with the uranium you are carrying I guess but you can carry a stack of 50 sheet so you'll never really run out). 
This mean that you now have to think about your ammo and get briefly to cover to reload as moving while reloading cancel it and give you more rads and some toxin damage immediately.

A cool thing about energy weapons is switching firemodes on the go, I gave it two additional firemode to make it a little bit more unique compared to regular egun and also enable greater warcrimes: irradiate and suffocate
-irradiate shoot an xray laser like the xray gun, 10 shots (xray gun is 14 so if you only want the xray and without fear of cancer then you tkae it over the aegun)
-suffocate bring once again my latest warcrime beams: anoxia rays! 8 oxygen damage, 20 shots

Idea is adding a decision between safe egun and powerful but dangerous aegun

It also make printing 3279432463809 aeguns be less meta since you will have to get rads for every shot because it start empty.

# Spriting
disable
nucgun disable

kill
nucgun kill

irradiate
nucgun irradiate

suffocate
nucgun suffocate

ingame
showcase

# Wiki Documentation
Advanced energy gun now use uranium sheets and ores to be reloaded, can't be recharged otherwise, ore is 50% less effective than sheets.

Start empty when printed.

Moving while recharging cancel the action and give you 5 toxin damage and 100 rads if you used uranium ore, 10 toxin and 200 rads if you used an uranium sheet.

-20 disable: 30 stamina disabler beam
-10 kill: 20 burn laser
-10 irradiate: 10 burn and 300 rads xray lasers
-20 suffocate: 8 oxygen anoxia rays

# Changelog

:cl:  
rscadd: Aegun can now shoot xray lasers and anoxia rays
tweak: Aegun now use uranium to recharge, doesn't self charge anymore and still do not fit in rechargers
/:cl:

### CODE REVIEW

**Comment by ynot01** [comment]
very cool

**Comment by cuackles** [comment]
could you show them ingame please, preferably all of them next to each other on the ground

**Comment by 00ze-cyclone** [comment]
> could you show them ingame please, preferably all of them next to each other on the ground

done

**Comment by cuackles** [comment]
could you lower the saturarion on the green and red one to make them just slightly darker

**Comment by 00ze-cyclone** [comment]
> could you lower the saturarion on the green and red one to make them just slightly darker

I'll try

File: code/modules/projectiles/guns/energy/energy_gun.dm [hunk 1]
```diff
@@ -130,63 +130,42 @@
 	trigger_guard = TRIGGER_GUARD_NONE
 	ammo_x_offset = 2
 
-/obj/item/gun/energy/e_gun/nuclear
+obj/item/gun/energy/e_gun/nuclear
```
Comment (Moltijoe) [inline code/modules/projectiles/guns/energy/energy_gun.dm]:
```suggestion
/obj/item/gun/energy/e_gun/nuclear
```

**Review by Moltijoe — CHANGES_REQUESTED** [review]

**Comment by MajManatee** [comment]
cool as it is, i cannot see anyone using a weapon that deliberately kills themselves.

**Comment by 00ze-cyclone** [comment]
> cool as it is, i cannot see anyone using a weapon that deliberately kills themselves.

There are people who play plasmaman to remove their helmet and set people on fire, we have TTV suicide bombers, I'm sure a lot of people would actually use a gun that kill the user slowly.

**Comment by Maia-J** [comment]
Preternis is the new sec race now it seems

**Comment by 00ze-cyclone** [comment]
let's see if those conflicts are fixed now

**Review by cuackles — APPROVED** [review]

**Comment by ThatLing** [comment]
Balance council approved

**Review by Moltijoe — APPROVED** [review]
completely forgot i had reviewed this

**Pull request merged**